nautobot/core/views/generic.py

```diff
@@ -1278,18 +1278,53 @@ def get_required_permission(self):
     def get(self, request):
         return redirect(self.get_return_url(request))
 
+    def handle_bulk_delete_all(self, request, **kwargs):
+        model = self.queryset.model
+
+        if self.filterset is not None:
+            queryset = self.filterset(request.GET, model.objects.only("pk")).qs
```

Comment:
Does calling a filterset on a `model.objects.only("pk")` require a bunch of extra DB lookups to resolve the other model fields relevant to the filters? I'd think so.

Reply:
Yea, that seems plausible